<launch>
  <node pkg="cortex_ros" type="stream" name="stream" output="screen">
  </node>
	<!-- Launches the AR.Drone driver -->
	<node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" clear_params="true">
	    <param name="outdoor" value="0" />				<!-- If we are flying outdoors, will select more aggressive default settings -->
	    <param name="flight_without_shell" value="0" /> <!-- Changes internal controller gains if we are flying without the propeller guard -->
	    
	    <param name="altitude_max" value="3000" />		<!-- in millimeters = 3 meters = 9' -->
        <param name="altitude_min" value="50" />		<!-- in millimeters = 5cm = 2" -->
        <param name="euler_angle_max" value="0.349066" />	<!-- maximum allowable body angle in radians = 5 degrees -->
        <param name="control_vz_max" value="1000" />		<!-- maximum z velocity in mm/sec = 0.2m/sec -->
	    <param name="control_yaw" value="3.490659" />		<!-- maximum rotation rate in radians/sec = 40 degrees per second (1/9 rev/sec) -->

	    <param name="detect_type" value="10" />
	    <param name="detections_select_h" value="32" />	<!-- In the horizontal camera, look for the orange-blue-orange markers -->
	    <param name="detections_select_v_hsync" value="128" />  <!-- In the vertical camera, detect the A4 marker -->
	    <param name="enemy_colors" value="3" />			<!-- Orange-Blue-Orange = 3 -->
	    <param name="enemy_without_shell" value="0" />	<!-- Detect the indoor stickers, rather than the outdoor hull -->
	</node>

<!-- run image_proc -->
<group ns="ardrone">
<node pkg="image_proc" type="image_proc" name="image_proc" />
</group>

	<node name="joy_node" pkg="joy" type="joy_node" output="screen" clear_params="true">
		<param name="dev" type="str" value="/dev/input/js0"/>
	</node>
<!-- Launches the joystick controller -->
	<node name="joystick_controller" pkg="quad_command" type="risc_controller.py" output="screen" required="true">
		
		<!-- Configures the joystick button mapping -->
		<param name="JoytoWay"	                value="0" />
		<param name="ButtonEmergency"	        value="2" />
		<param name="TakeoffLand"	        value="6" />
		<param name="Up"	         	value="9" />
		<param name="Down"			value="10" />
		<param name="ShutDownNode"		value="7" />
		<param name="Quad1"	         	value="3" />
		<param name="Quad2"	         	value="1" />
		<param name="Quad3"	         	value="4" />
		<param name="Quad4"	         	value="5" />
		

		<!-- Configures the joystick axis mapping -->
		<param name="AxisRoll"			value="0" />
		<param name="AxisPitch"			value="1" />
		<param name="AxisYaw"			value="3" />
		<param name="Waynorthsouth"	       	value="6" />
		<param name="Wayeastwest"	       	value="7" />

		<!-- Configures the joystick mapping -->
		<param name="ScaleRoll"			value="10" />
		<param name="ScalePitch"		value="10" />
		<param name="ScaleYaw"			value="1" />
		<param name="ScaleZ"			value="1" />

	</node>
    <node name="transforms" pkg="quad_command" type="transforms"  output="screen" required="true">
    </node>

    <node name="quad_track_cortex" pkg="risc_visual" type="quad_track_cortex"  output="screen" required="true">
    </node>

</launch>
